xtevent/_eventgenvars.ado

```diff
@@ -220,6 +224,12 @@ program define _eventgenvars, rclass
 		di "If event dummies and variables are saved, event-time will be missing."	
 		loc impute =""
 	}
+	
+	if `bin'==0 & "`w_type'"=="string" {
+		di "The policy variable is not binary. Assuming non-staggered adoption."
+		di as err _n "Option {bf:window(`lwindow')} can be used only if policyvar follows staggered adoption."

```

Comment:
@Constantino-Carreto-Romero this is only true with a non-binary policy variable. With a binary policy variable window can be used with impute(nuchange) even without staggered adoption, right? Modify the error message accordingly.

Reply:
@jorpppp thanks
The error message will be one of:
- Option `window(max)` can be used only if policyvar follows staggered adoption
-  Option `window(balanced)` can be used only if policyvar follows staggered adoption

because at this point in `_eventgenvars` `lwindow` (and `rwindow`) equals `max` or `balanced` if the local `w_type` equals `string`. 
So I think the error message is fine, but let me know if I'm missing something.

Reply:
@Constantino-Carreto-Romero The bigger question is, why can't you use `window(max)` or `window(balanced)` outside of staggered adoption?

Reply:
@Constantino-Carreto-Romero We will only enable these options under staggered adoption so the message is fine.